connectivity/tests/lrp.go

```diff
@@ -0,0 +1,190 @@
+
+	for _, policy := range t.CiliumLocalRedirectPolicies() {
+		spec := policy.Spec
+		if spec.RedirectFrontend.AddressMatcher == nil {
+			continue
+		}
+		policies = append(policies, policy)
+		frontend := check.NewLRPFrontend(spec.RedirectFrontend)
+		frontendStr := net.JoinHostPort(frontend.Address(features.IPFamilyV4), fmt.Sprint(frontend.Port()))
+		lrpBackendsMap := make(map[string][]string)
+		// Check for LRP backend pods deployed on nodes in the cluster.
+		for _, pod := range t.Context().LrpBackendPods() {
+			node := pod.NodeName()
+			podIP := pod.Pod.Status.PodIP
+			if _, ok := lrpBackendsMap[node]; !ok {
+				lrpBackendsMap[node] = []string{podIP}
+			}
+			lrpBackendsMap[node] = append(lrpBackendsMap[node], podIP)
+		}
+		// Wait until the local redirect entries are plumbed in the BPF LB map
+		// on the cilium agent nodes hosting LRP backend pods.
+		WaitForLocalRedirectBPFEntries(ctx, t, frontendStr, lrpBackendsMap)
+	}
+
+	// Tests client pods to LRP frontend connectivity: traffic gets redirected
+	// to the LRP backends.
+	for _, pod := range t.Context().LrpClientPods() {
+		pod := pod
+
+		for _, policy := range policies {
+			policy := policy
+
+			if policy.Spec.SkipRedirectFromBackend != s.skipRedirectFromBackend {
+				continue
+			}
+
+			i := 0
+			lf := check.NewLRPFrontend(policy.Spec.RedirectFrontend)
+			t.NewAction(s, fmt.Sprintf("curl-%d", i), &pod, lf, features.IPFamilyV4).Run(func(a *check.Action) {
+				a.ExecInPod(ctx, ct.CurlCommand(lf, features.IPFamilyV4))
+				i++
+			})
+		}
+	}
+
+	// Tests LRP backend pods to LRP frontend connectivity: traffic gets redirected
+	// based on the configured skipRedirectFromBackend flag.
+	for _, pod := range t.Context().LrpBackendPods() {
+		pod := pod
+
+		for _, policy := range policies {
+			policy := policy
+
+			if policy.Spec.SkipRedirectFromBackend != s.skipRedirectFromBackend {
+				continue
+			}
+
+			i := 0
+			lf := check.NewLRPFrontend(policy.Spec.RedirectFrontend)
+			t.NewAction(s, fmt.Sprintf("curl-%d", i), &pod, lf, features.IPFamilyV4).Run(func(a *check.Action) {
+				a.ExecInPod(ctx, ct.CurlCommand(lf, features.IPFamilyV4))
+
+				if policy.Spec.SkipRedirectFromBackend {
+					a.ValidateFlows(ctx, pod, a.GetEgressRequirements(check.FlowParameters{
+						AltDstIP:   lf.Address(features.IPFamilyV4),
+						AltDstPort: lf.Port(),
+					}))
+				}
+				i++
+			})
+		}
+
+	}
+}
+
+func WaitForLocalRedirectBPFEntries(ctx context.Context, t *check.Test, frontend string, backendsMap map[string][]string) {
+	ct := t.Context()
+	w := wait.NewObserver(ctx, wait.Parameters{Timeout: 20 * time.Second})
+	defer w.Cancel()
+
+	ensureBPFLBEntries := func() error {
+		for _, ciliumPod := range ct.CiliumPods() {
+			node := ciliumPod.Pod.Spec.NodeName
+			backends, ok := backendsMap[node]
+			if !ok {
+				// No LRP backend pods deployed on this node.
+				continue
+			}
+			cmd := strings.Split("cilium bpf lb list -o json", " ")
```

Comment:
nit: The command is invariant might as well move it out of the loop and split once